features/search-replace.feature

```diff
@@ -1106,6 +1106,28 @@ Feature: Do global search/replace
       a:1:{i:0;O:10:"CornFlakes":0:{}}
       """
 
+  Scenario: Warn and ignore type-hinted objects that have some error in deserialization
+
+    Given a WP install
+    And I run `wp db query "INSERT INTO wp_options (option_name,option_value) VALUES ('cereal_isation','O:13:\"mysqli_result\":5:{s:13:\"current_field\";N;s:11:\"field_count\";N;s:7:\"lengths\";N;s:8:\"num_rows\";N;s:4:\"type\";N;}')"`
+    And I run `wp db query "INSERT INTO wp_options (option_name,option_value) VALUES ('cereal_isation_2','O:8:\"mysqli_result\":5:{s:13:\"current_field\";i:1;s:11:\"field_count\";i:2;s:7:\"lengths\";a:1:{i:0;s:4:\"blah\";}s:8:\"num_rows\";i:1;s:4:\"type\";i:2;}')"`
+
+    When I try `wp search-replace mysqli_result stdClass`
+    Then STDERR should contain:
+      """
+      Warning: Skipping an inconvertible serialized object: "O:13:"mysqli_result":5:{s:13:"current_field";N;s:11:"field_count";N;s:7:"lengths";N;s:8:"num_rows";N;s:4:"type";N;}", replacements might not be complete.
+      """
+    And STDOUT should contain:
+      """
+      Success: Made 1 replacement.
+      """
+
+    When I run `wp db query "SELECT option_value from wp_options where option_name='cereal_isation_2'"`
+    Then STDOUT should contain:
+      """
+      O:8:"stdClass":5:{s:13:"current_field";i:1;s:11:"field_count";i:2;s:7:"lengths";a:1:{i:0;s:4:"blah";}s:8:"num_rows";i:1;s:4:"type";i:2;}
+      """
```

Comment:
When we validate the second object, we should also make sure it deserializes correctly (and that we didn't break serialization).

Reply:
makes sense, added it into behat. Asserted the replace was successful and a couple of other fields are what we expect